internal/network/machine_network_cidr.go

```diff
@@ -115,12 +115,25 @@ func VerifyVipFree(hosts []*models.Host, vip string, machineNetworkCidr string,
 	return IpInFreeList(hosts, vip, machineNetworkCidr, log)
 }
 
-func VerifyVip(hosts []*models.Host, machineNetworkCidr string, vip string, vipName string, verification *models.VipVerification, log logrus.FieldLogger) (models.VipVerification, error) {
-	if machineNetworkCidr == "" {
+func VerifyVip(hosts []*models.Host, machineNetworks []*models.MachineNetwork, vip string, vipName string, verification *models.VipVerification, log logrus.FieldLogger) (models.VipVerification, error) {
+	machineNetworkCidr := ""
+	validMachineNetwork := false
+	for _, machineNetwork := range machineNetworks {
+		if machineNetwork == nil || machineNetwork.Cidr == "" {
+			continue
+		}
+		validMachineNetwork = true
+		if ipInCidr(vip, string(machineNetwork.Cidr)) {
+			machineNetworkCidr = string(machineNetwork.Cidr)
+			break
+		}
+	}
+
+	if !validMachineNetwork {
 		return models.VipVerificationUnverified, errors.Errorf("%s <%s> cannot be set if Machine Network CIDR is empty", vipName, vip)
 	}
-	if !ipInCidr(vip, machineNetworkCidr) {
-		return models.VipVerificationFailed, errors.Errorf("%s <%s> does not belong to machine-network-cidr <%s>", vipName, vip, machineNetworkCidr)
+	if machineNetworkCidr == "" {
```

Comment:
This test is irrelevant. The machine network cannot be valid and empty.

Reply:
Yes it can - validMachineNetwork will be true if any machine networks are defined, but machineNetworkCidr will only be set if the VIP is in one of them.

Reply:
This is maintaining the previous behaviour. If there are no machineNetworks specified, we return `VipVerificationUnverified`. If there are machineNetworks specified and the VIP isn't in any of them, we return `VipVerificationFailed`.

I think what's confusing in the diff is that `machineNetworkCidr` was previously a parameter, and now it's a local variable.